kubespawner/spawner.py

```diff
@@ -345,6 +355,29 @@ def _namespace_default(self):
         """
     )
 
+    secret_name_template = Unicode(
+        'jupyter-{username}{servername}',
+        config=True,
+        help="""
+        Template to use to form the name of user's secret.
+
+        `{username}` is expanded to the escaped, dns-label safe username.
+
+        This must be unique within the namespace the pvc are being spawned
+        in, so if you are running multiple jupyterhubs spawning in the
+        same namespace, consider setting this to be something more unique.
+        """
+    )
+
+    internal_ssl_directory = Unicode(
+        None,
+        allow_none=True,
+        config=True,
+        help="""
+        Location of Jupyter's internal ssl directory.
+        """
+    )
```

Comment:
Could we provide a default value for this? Is there a reason not to?

Reply:
This can be determined from the Jupyter Hub environment variable which is what I was previously doing, but @yuvipanda asked me to not do that.

If we don't want to use the jupyterhub environment variable I don't believe it makes sense for us to attempt a default.

Reply:
Hmmm, I'm trying to put pieces together still. Could you help me out by making it more explicit what this traitlet configures?

- Is this a location on the JupyterHub pod or singleuser pod?
- What certificates (and keys?) will be contained here?

I'll open a new comment where I summarize my current understanding, perhaps that is better starting point.

Reply:
> Is this a location on the JupyterHub pod or singleuser pod?

It is the location on the jupyterhub pod.

> What certificates (and keys?) will be contained here?

All certificates, for users and for JH, i think.